docs/testing-styleguide.MD

```diff
@@ -0,0 +1,278 @@
+
+### Functional Browser Testing
+#### Manual Testing
+Every new or amended component should be fully tested in either Storybook and/or [codesandbox.io](https://codesandbox.io) using functional testing, UI testing and internationalization testing techniques. We use [Jira](https://www.atlassian.com/software/jira) for issue and project tracking. 
+
+##### Browser Support
+Carbon is supported in the following browsers:
+* Chrome
+* Firefox
+* Safari
+* Edge Chromium
+
+Components should be tested in all supported browsers unless the piece of functionality is not specifically related to a UI feature, e.g. decimal precision.
+
+##### Jira
+For every testable task, a test session needs to be created for each of the supported browsers. Results should be added to each of the conditions in the created test sessions with a pass `(/)` or fail `(x)` mark.
+
+Any issues found during testing should be raised as a "Defect" in Jira unless the ticket is already of a “Bug” type, in which case comments and screenshots/recordings/GIFs can be added to the existing ticket.
+
+If the issue is present on the master branch you should first check if the issue has already been logged in GitHub. If it is present, it should be upvoted with a “+1” comment. If it has not been logged, details of how to do this are contained in the [CONTRIBUTING.MD](https://github.com/Sage/carbon/blob/master/CONTRIBUTING.md) guide. Collaborators internal to Sage should also log a Jira ticket for the issue as a "Bug" type with the GitHub issue linked. However, there should be no link from the GitHub issue to the Jira ticket.
+
+##### Storybook
+Storybook forms our Carbon component library documentation. The component stories are written in `MDX` format containing as many examples as required to showcase our components and with minimal knobs for changing prop states. Every component should have at least one story (named `default`/`multiple`/`validation` etc.) located in the `Design System` directory with `Docs`, `Notes` and `Accessibility` sections. Where required, it should be possible to test all events, e.g. `onClick`, `drag` etc, in these stories. Results of events tests can be found in the `Actions` tab.
+
+Where the stories in the main `Design System` documentation directory do not cover all scenarios required to fully test the component, additional stories must be created in the `Test` directory. For example, scenarios testing special characters, all combinations of size/type/state etc should have stories in the `Test` directory under the relevant component and named `visual`. A developer may need to assist with writing the additional required stories here.
+
+To clarify the purpose of each directory,
+
+| Directory     | Usage                                                                    |
+|---------------|--------------------------------------------------------------------------|
+| Design System | User facing documentation, primary visual regression tests               |
```

Comment:
I think that whole `visual regression` should be done in one place for the relevant component `Test > [component name] > visual`. We shouldn't split it into two separate places. I mean `Test > [component name] > visual` should contain as many examples as required for visual regression without any additional description

Reply:
I thought we agreed though that we didn't want to duplicate stories just for testing purposes? If we can do what visual testing we need in in the canvas in `Design System` then do we really need to duplicate the story just to run visual comparison testing? May be I missed something during the team call about this so happy to change it if that is the case.

Reply:
During the team meeting we had about the Applitools strategy, we discussed how that approach would mean duplicating stories just for visual testing purposes. We agreed to do visual testing against the Docs in stories in `Design System` unless we required specific test scenarios that were not covered in `Design System` or if we find that the length of the page in Docs is too long and contains too much text and white space. Happy to discuss it again though? May be when we look at the numbers of components requiring visual testing to get an estimate of required snapshots for the next meeting with Applitools we can get an idea of how much duplication there'd be if we took this approach? If we change our processes then I can change this doc then.

Reply:
Since decided to place all visual testing stories in the `Test` directory as you mentioned above. Updated this doc to reflect those discussions.